weather_calculation.py

```diff
@@ -0,0 +1,25 @@
+class WeatherCalculations:
+    def update_calculations(self, reading):
+        if not hasattr(self, "num_readings"):
+            self.highest_temp = self.lowest_temp = self.highest_humidity = reading
+            self.total_max_temp = reading.max_temp
+            self.total_min_temp = reading.min_temp
+            self.total_mean_humidity = reading.mean_humidity
+            self.num_readings = 1
+        else:
+            self.highest_temp = max(self.highest_temp, reading, key=lambda x: x.max_temp)
```

Comment:
Why we used lambda here?

Reply:
Lambda makes our coding precise. Instead of calling a separate function to get the key which is max_temp here, we can use lambda function. We are comparing two objects on basis of this key and we cant pass the value directly to it as 'key' expects a function with a single argument